North-Seattle-College/ad440-winter2021-thursday-repo
# Feature/team1 issue71 complete azure users user id function

**Resolves Task/BUG** :[#71](https://github.com/North-Seattle-College/ad440-winter2021-thursday-repo/issues/71)

**To Test API function users/{userId}** 
- Test this function on Azure:
- API call - [users/1](https://nsc-functionsapp-team1.azurewebsites.net/api/users/1)


- This code completes the implementation of the Azure function for the previous PR #49  that is already merged into the dev branch
           - The following methods got implemented and Updated: GET/POST/PUT/PATCH/DELETE

- Once this PR gets approved the function will be deployed automatically into the development branch
-  To test this feature locally make sure to have the following: 
      - Azure function extension for VSCode
      - Pull the branch to your machine and make sure to be on the correct branch
      - Install the ODBC driver so you can connect to the database: For any issues with the ODBC driver checkout PR49 for additional documentation to fix ODBC driver
      -In local.settings.json at the root of the functions folder, enter in the values for the keys below in the following format: { "Values": { "KEY": "VALUE" } }
ENV_DATABASE_NAME (name of the database)
ENV_DATABASE_PASSWORD
ENV_DATABASE_SERVER (this is the server location)
ENV_DATABASE_USERNAME
- You can click F5 or click run->Start Debugging in the functions folder to start it up

- The GET request takes the userId and returns all a specific user by their Id. 
- The POST request is not allowed for this call.
- The PUT request takes a JSON object { "firstName": "add your firstname", "lastName": "add your last name", "email": "addYourEmail@email.com" } and will update the user firstName, lastName, email when you select a specific userId. 
- The DELETE request takes the userId and deletes the user from the database.
-To test using JSON, copy the link provided by the debugger (the "/users/{userId}" ) into Postman, and use as usual for GET and PUT (info goes into the body), DELETE (userId). PUT and PATCH are not allowed
- Error handling and Debug handling included in this feature





DATE | ACTIVITY | TIME
-- | -- | --
1/21 | Review PR49 after class to get an idea of what I need to update | 60 minutes
1/23 | More research of pyodbc Crud functionality | 90 minutes
1/24 | encounter issue with running function locally | 120 minutes
1/24 | Still figuring out what was going on with the issue| 60 minutes
1/24 | Started working on functionality| 60 minutes
1/25 | Setup all rest methods (get, post, put, delete) to work as expected | 120 minutes
1/26 | Added error handling to post requests | 90 minutes
1/26 | Added storage account to function app on azure | 20 min
1/28 | Setup all rest methods (get, post, put, delete, patch) to work as expected | 230 minutes
1/28 | Added more error handling for the DB connection logging | 60 minutes
1/28 | wrote this PR which I will have to update again |60 minutes
